### PR TITLE
shakemaps: use relative paths

### PR DESCRIPTION
**Pull request by schmidni**
With this method it is possible to use different local pathnames for the shakemap files:

```
"grid_url": "shakemap_files/grid.xml"
"grid_url": "grid.xml"
"grid_url": /home/user/calculation/grid.xml
"grid_url": file:/home/user/calculation/grid.xml
```

`pathlib.Path` accepts relative and absolute paths, `pathname2url` makes sure it is converted into the right form to be used by `urlopen` independently from OS.

### CODE REVIEW

File: openquake/hazardlib/shakemap/parsers.py [hunk 1]
```diff
@@ -83,17 +84,36 @@ def urlextract(url, fname):
...
+    If a relative path is given for the file, parse it so it can be
+    read with 'urlopen'.
+    :param url: path/url to be parsed
+    """
+    if not url.startswith('file:') and not url.startswith('http'):
+        file = pathlib.Path(url)
+        if file.is_file():
+            return 'file:{}'.format(pathname2url(str(file.absolute())))
+        else:
+            raise RuntimeError(
```
Comment (micheles) [inline openquake/hazardlib/shakemap/parsers.py:98]: Should be a FileNotFoundError

File: openquake/hazardlib/shakemap/parsers.py [hunk 2]
```diff
@@ -83,17 +84,36 @@ def urlextract(url, fname):
                 raise
 
 
+def parse_url_path(url):
```
Comment (micheles) [inline openquake/hazardlib/shakemap/parsers.py:87]: I would call this path2url

**Comment by micheles** [comment]
LGTM apart from the minor comments

File: openquake/hazardlib/shakemap/parsers.py (hunk 2 above)
Comment (schmidni) [reply openquake/hazardlib/shakemap/parsers.py]: good idea

File: openquake/hazardlib/shakemap/parsers.py (hunk 1 above)
Comment (schmidni) [reply openquake/hazardlib/shakemap/parsers.py]: corrected

**Pull request merged**